src/screens/Settings/SettingsScreen.js

```diff
@@ -0,0 +1,179 @@
+/*
+ * @author Arthur Martello <arthur.martello@etu.utc.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GPL-3.0
+ */
+
+import React from 'react';
+import { Alert, RefreshControl, ScrollView, Text, View } from 'react-native';
+import Ionicons from '@expo/vector-icons/Ionicons';
+import { connect } from 'react-redux';
+import colors from '../../styles/colors';
+import BlockTemplate from '../../components/BlockTemplate';
+import { _, Settings as t } from '../../utils/i18n';
+import SwitchBlockTemplate from '../../components/SwitchBlockTemplate';
+import { Config, PayUTC } from '../../redux/actions';
+import PayUTCApi from '../../services/PayUTC';
+
+class SettingsScreen extends React.PureComponent {
+	static navigationOptions = {
+		title: t('title'),
+		header: null,
+		headerForceInset: { top: 'never' },
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = { expectedStatus: false };
+
+		this.onLockChange = this.onLockChange.bind(this);
+	}
+
+	componentDidMount() {
+		this.onRefresh();
+	}
+
+	onRefresh() {
+		const { lockStatusFetching, lockStatus, dispatch } = this.props;
+
+		if (!lockStatusFetching) {
+			dispatch(PayUTC.getLockStatus());
+			this.setState({ expectedStatus: !lockStatus });
+		}
+	}
+
+	onLockChange(value) {
+		const { dispatch, lockStatusFetching } = this.props;
+
+		if (lockStatusFetching) {
+			return;
+		}
+
+		dispatch(
+			Config.spinner({
+				visible: true,
+				textContent: value ? t('locking') : t('unlocking'),
+			})
+		);
+
+		PayUTCApi.setLockStatus(value).then(([status]) => {
+			dispatch(
+				Config.spinner({
+					visible: false,
+				})
+			);
+
+			if (status !== true && status !== false) {
```

Comment:
If think you should do it in the catch method.
The API will always return the good response

Reply:
Do not forget to set the spinner no more visible if you catch an error :p

Reply:
Catch is never thrown with this API request